sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/TracerProviderConfiguration.java

```diff
@@ -78,16 +80,17 @@ static SdkTracerProvider configureTracerProvider(
 
     if (!exportersByNameCopy.isEmpty()) {
       SpanExporter compositeSpanExporter = SpanExporter.composite(exportersByNameCopy.values());
-      spanProcessors.add(configureBatchSpanProcessor(config, compositeSpanExporter));
+      spanProcessors.add(configureBatchSpanProcessor(config, compositeSpanExporter, meterProvider));
     }
 
     return spanProcessors;
   }
 
   // VisibleForTesting
   static BatchSpanProcessor configureBatchSpanProcessor(
-      ConfigProperties config, SpanExporter exporter) {
+      ConfigProperties config, SpanExporter exporter, MeterProvider meterProvider) {
     BatchSpanProcessorBuilder builder = BatchSpanProcessor.builder(exporter);
+    builder.setMeterProvider(meterProvider);
```

Comment:
#nit: slight preference for the following:
```suggestion
    BatchSpanProcessorBuilder builder = BatchSpanProcessor.builder(exporter).setMeterProvider(meterProvider);
```